lib/analist/annotations.rb

```diff
@@ -97,7 +120,12 @@ def send_annotations # rubocop:disable Metrics/MethodLength
         where: lambda do |receiver_return_type|
                  Annotation.new(
                    { type: receiver_return_type[:type], on: :collection },
-                   Set.new([[], [Hash]]),
+                   Set.new([[],
+                            [Array],
+                            [String, Hash],
+                            [String],
+                            [String, String],
+                            [String, String, String]]), # TODO: support countless arguments
```

Comment:
Maybe this could be solved by using an [infinite sequence Enumerator](https://rossta.net/blog/infinite-sequences-in-ruby.html). Of course you'd have to take the arity into account to prevent endless loops 🙂

Reply:
Good catch!